packages/phpstan-rules/tests/Rules/Explicit/SameNamedParamFamilyRule/SameNamedParamFamilyRuleTest.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Symplify\PHPStanRules\Tests\Rules\Explicit\SameNamedParamFamilyRule;
+
+use Iterator;
+use PHPStan\Rules\Rule;
+use Symplify\PHPStanExtensions\Testing\AbstractServiceAwareRuleTestCase;
+use Symplify\PHPStanRules\Rules\Explicit\SameNamedParamFamilyRule;
+
+/**
+ * @extends AbstractServiceAwareRuleTestCase<SameNamedParamFamilyRule>
+ */
+final class SameNamedParamFamilyRuleTest extends AbstractServiceAwareRuleTestCase
+{
+    /**
+     * @dataProvider provideData()
+     * @param array<string|int> $expectedErrorMessagesWithLines
+     */
+    public function testRule(string $filePath, array $expectedErrorMessagesWithLines): void
+    {
+        $this->analyse([$filePath], $expectedErrorMessagesWithLines);
+    }
+
+    public function provideData(): Iterator
+    {
+        yield [__DIR__ . '/Fixture/SkipSomeClassWithoutParents.php', []];
```

Comment:
wasnt there a way to yield all fixtures from a dir instead of yielding every file hard-coded?

Reply:
How would you separate those with error and without errors?

Reply:
Maybe based on file-name prefix (like you already have a "skip" prefix)

Reply:
That could work. What about messages with values, e.g. like those you've recommended here?

Reply:
I think it would be a first step to load fixtures by file prefix. 

Those which need additional error message could still be added like we do today